### PR TITLE
UOE-4962: multi-currency support, add bid.ext.origbidcpmusd field

### PR DESCRIPTION
**Pull request by pm-nilesh-chate**
# Description

Sample request:
```
{
    "cur": ["GBP"],
}
```

Sample response:
```
{
    "seatbid": [
        {
            "bid": [{
                    "price": 1.0742528068018218,
                    "ext": {
                        "origbidcpm": 100,
                        "origbidcpmusd": 1.2154054337953153,
                        "origbidcur": "INR",
                     }
                }]
            "seat": "pubmatic"
        }
    ],
    "cur": "GBP"
}
```

# Checklist:

- [x] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [x] JIRA number is added in the PR title and the commit message.
- [x] Updated the `header-bidding` repo with appropiate commit id.
- [x] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

File: exchange/bidder.go [hunk 1]
```diff
@@ -356,9 +376,11 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, bidderRequest Bidde
 						}
 
 						originalBidCpm := 0.0
+						originalBidCPMUSD := 0.0
 						if bidResponse.Bids[i].Bid != nil {
 							originalBidCpm = bidResponse.Bids[i].Bid.Price
 							bidResponse.Bids[i].Bid.Price = bidResponse.Bids[i].Bid.Price * adjustmentFactor * conversionRate
+							originalBidCPMUSD = originalBidCpm * adjustmentFactor * conversionRateUSD
```
Comment (ashishshinde-pubm) [inline exchange/bidder.go:383]: can we replace - 
originalBidCPMUSD = originalBidCpm * adjustmentFactor * conversionRateUSD
with
originalBidCPMUSD = bidResponse.Bids[i].Bid.Price

Comment (pm-nilesh-chate) [reply exchange/bidder.go:383]: `bidResponse.Bids[i].Bid.Price` could be in `INR` and `originalBidCPMUSD` in `USD`. i.e they both could be different, so cannot assign them to one another.

File: exchange/bidder.go [hunk 2]
```diff
@@ -285,10 +294,21 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, bidderRequest Bidde
...
 				}
 
+				// no need of conversionRateUSD if
+				// - bids with conversionRate = 0 would be a dropped
+				// - response would be in USD
+				if conversionRate != float64(0) && selectedCur != "USD" {
+					conversionRateUSD, err = conversions.GetRate(bidResponse.Currency, "USD")
+					if err != nil {
+						errs = append(errs, fmt.Errorf("failed to get USD conversion rate for WL and WTK %v", err))
+					}
```
Comment (PubMatic-OpenWrap) [inline exchange/bidder.go:309]: is it required to mention WL and WTK in error message ?

File: exchange/exchange.go [hunk 3]
```diff
@@ -1139,6 +1139,11 @@ func makeBidExtJSON(ext json.RawMessage, prebid *openrtb_ext.ExtBidPrebid, impEx
 		extMap[openrtb_ext.OriginalBidCurKey] = originalBidCur
 	}
 
+	//ext.origbidcpmusd
+	if originalBidCpmUSD > 0 {
```
Comment (PubMatic-OpenWrap) [inline exchange/exchange.go:1143]: originalBidCpmUSD > float(0)

File: exchange/bidder.go (hunk 2 above)
Comment (pm-nilesh-chate) [reply exchange/bidder.go:309]: added to differential failure for price conversions between bid.price and price needed for WL&WTK

**Review by pm-isha-bharti — APPROVED** [review]

File: exchange/bidder.go (hunk 1 above)
Comment (pm-viral-vala) [inline exchange/bidder.go:383]: is there any scenario where originalBidCPMUSD value will be zero for valid bid? if yes then what are expectations from OW and analytics?

**Review by pm-viral-vala — APPROVED** [review]
LGTM

File: exchange/bidder.go (hunk 1 above)
Comment (pm-nilesh-chate) [reply exchange/bidder.go:383]: yes, `originalBidCPMUSD` could be 0 if USD conversion rates are not available. Although, how often would it be?.
Anyway, please refer https://git.pubmatic.com/PubMatic/header-bidding/pull/910/files#r74097 for its action items

**Pull request merged**